src/d3a/models/area/__init__.py

```diff
@@ -277,10 +277,14 @@ def tick(self):
                     market.match_offers_bids()
 
         self.events.update_events(self.now)
+
+    def update_area_current_tick(self):
```

Comment:
Can you please explain why you are calling the `update_area_current_tick` from the simulation.py instead of from the area tick method?

Reply:
Since the ticks are dispatched from bottom to top, current_tick was getting updated more than once per area for an intermediate area having multiple children.

Reply:
Ahhh, I understand now. Nice catch!